js/app.js

```diff
@@ -1,5 +1,66 @@
-angular.module('GiphyApp', [])
+GiphyApp = angular.module('GiphyApp', ['infinite-scroll'])
 
-.run(function() {
-  console.log('App has loaded!');
-});
+// GiphyApp.run(function() {
+//   console.log('App has loaded!');
+//   Populate();
+// });
+
+GiphyApp.controller('GifSearch', ['$scope', '$http', function($scope, $http) {
+
+  Populate();
+
+  $scope.search = function(searchTerm) {
+    $scope.term = searchTerm
+    Search();
+  }
+
+  function Populate() {
+    $http.get('http://api.giphy.com/v1/stickers/search', {
+      params: {
+        api_key: 'dc6zaTOxFJmzC',
+        q: 'pokemon',
+        limit: 6,
+        offset: 20,
+        rating: 'y'
+      }
+    })
+    .then(function success(response) {
+      $scope.gifs = response.data.data
+    }, function error(response) {
+      console.log(response);
+    })
+  }
+
+  function Search() {
+    $http.get('https://api.giphy.com/v1/gifs/search', {
+      params: {
+        api_key: 'dc6zaTOxFJmzC',
+        q: $scope.term,
+        limit: 24
+      }
+    })
+    .then(function success(response) {
+      $scope.gifs = response.data.data
+    }, function error(response) {
+      console.log(response);
+    })
+  }
+
+  function Search2() {
+    $http.get('https://api.giphy.com/v1/gifs/search', {
+      params: {
+        api_key: 'dc6zaTOxFJmzC',
+        q: $scope.term,
+        limit: 24,
+        offset: 24
+      }
+    })
+    .then(function success(response) {
+      $scope.gifs = response.data.data
+    }, function error(response) {
+      console.log(response);
+    })
+  }
+
```

Comment:
Just for funsies, here is some code from my deliverable to handle the infinite scrolling:

    $scope.search = function () {
        if ($scope.searchTerm != $scope.oldSearchTerm) {$scope.giphys = [];};
        var req = {
            url: "http://api.giphy.com/v1/gifs/search",
            method: 'GET',
            params: {
                api_key: $scope.api,
                offset: $scope.offset,
                q: $scope.searchTerm,
            }
        }
        $http(req).then(function success(res) {
            $scope.giphys = $scope.giphys.concat(res.data.data);
        }, function error (res) {
            console.log(res);
        });
    };

    $scope.myPagingFunction = function (){
        $scope.oldSearchTerm = $scope.searchTerm;
        $scope.offset = 25 * $scope.count;
        $scope.search();
        $scope.count++;
    };